File: Backend/src/main/java/com/taskmanager/Task_Management_Application/service/project/ProjectService.java
package com.taskmanager.Task_Management_Application.service.project;

import com.taskmanager.Task_Management_Application.dto.ProjectDto;

public interface ProjectService {

    java.util.List<ProjectDto> getAllProjects();

    ProjectDto getProjectById(Long id);
    
    ProjectDto createProject(ProjectDto projectDto);

    ProjectDto updateProject(Long id, ProjectDto projectDto);

    void deleteProject(Long id);

    void assignMemberToProject(Long projectId, Long userId);

    void removeMemberFromProject(Long projectId, Long userId);
}

File: Frontend/src/services/project.service.js
import api from '../lib/axios';

export const ProjectService = {
  getProjects: async () => {
    const response = await api.get('/projects');
    return response.data;
  },

  getProjectById: async (id) => {
    const response = await api.get(`/projects/${id}`);
    return response.data;
  },

  createProject: async (projectData) => {
    const response = await api.post('/projects', projectData);
    return response.data;
  },

  updateProject: async (id, projectData) => {
    const response = await api.put(`/projects/${id}`, projectData);
    return response.data;
  },

  deleteProject: async (id) => {
    const response = await api.delete(`/projects/${id}`);
    return response.data;
  },

  assignMemberToProject: async (projectId, userId) => {
    const response = await api.put(`/projects/${projectId}/members/${userId}`);
    return response.data;
  },

  removeMemberFromProject: async (projectId, userId) => {
    const response = await api.delete(`/projects/${projectId}/members/${userId}`);
    return response.data;
  }
};
